diff --git a/include/stdlib.h b/include/stdlib.h
index 42ca8336..4ca2b39e 100644
--- a/include/stdlib.h
+++ b/include/stdlib.h
@@ -156,6 +156,8 @@ struct __locale_struct;
 float strtof_l(const char *__restrict, char **__restrict, struct __locale_struct *);
 double strtod_l(const char *__restrict, char **__restrict, struct __locale_struct *);
 long double strtold_l(const char *__restrict, char **__restrict, struct __locale_struct *);
+long long strtoll_l(const char *__restrict, char **__restrict, int, struct __locale_struct *);
+unsigned long long strtoull_l(const char *__restrict, char **__restrict, int, struct __locale_struct *);
 #endif
 
 #if defined(_LARGEFILE64_SOURCE) || defined(_GNU_SOURCE)
diff --git a/src/stdlib/strtol.c b/src/stdlib/strtol.c
index bfefea69..c1600307 100644
--- a/src/stdlib/strtol.c
+++ b/src/stdlib/strtol.c
@@ -48,9 +48,21 @@ uintmax_t strtoumax(const char *restrict s, char **restrict p, int base)
 	return strtoull(s, p, base);
 }
 
+long long strtoll_l(const char *restrict s, char **restrict p, int base, struct __locale_struct *l)
+{
+	return strtoll(s, p, base);
+}
+
+unsigned long long strtoull_l(const char *__restrict s, char **__restrict p, int base, struct __locale_struct *l)
+{
+	return strtoull(s,p,base);
+}
+
 weak_alias(strtol, __strtol_internal);
 weak_alias(strtoul, __strtoul_internal);
 weak_alias(strtoll, __strtoll_internal);
 weak_alias(strtoull, __strtoull_internal);
+weak_alias(strtoll_l, __strtoll_l_internal);
+weak_alias(strtoull_l, __strtoull_l_internal);
 weak_alias(strtoimax, __strtoimax_internal);
 weak_alias(strtoumax, __strtoumax_internal);
